### ISSUE TITLE
"error: Metricbeat FAILED" errors under Logs tab on enrolling 7.11 Windows agent with 7.10 default policy on upgraded kibana(7.10.2-7.11.0).

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#23812]
**Kibana version:** Kibana: 7.11.0 BC-5 Cloud environment

**Host OS and Browser version:** Windows 10, All

**Preconditions:**

1. 7.10.2 Cloud environment must be upgraded to 7.11.0.

**Build Details:**
```
    Artifact link used: https://staging.elastic.co/7.11.0-903dc0b6/downloads/beats/filebeat/filebeat-7.11.0-windows-x86_64.zip
    BUILD: 37827
    COMMIT: d801a7bb08e17368584d00fbd97a5d0006285b51
```

**Steps to reproduce:**
1. Login to Kibana cloud environment.
2. Install 7.11 agent using policy existing earlier on upgraded Kibana.
3. Navigate to Fleet>agent "Logs" tab.
4. Notice, agent is in "unhealthy" status.
5. Observe, "error: Metricbeat FAILED" is displayed in Logs tab. 

**Expected Result:**
 7.11 agent must be deployed successfully with 7.10 default policy on upgraded kibana cloud environment(7.10.2-7.11.0).

**Screenshots:**
![Upgrade](https://user-images.githubusercontent.com/77374876/106610431-ca5f4480-658c-11eb-9266-b753b5959d85.png)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review. Thanks

**Comment by manishgupta-qasource** [comment]
Reviewed & Assigned to @EricDavisX

**Comment by EricDavisX** [comment]
@amolnater-qasource can you please confirm 
 - which (specific) windows OS this is on? 
 - And which version of the package is on the system you are testing - is it 0.10.7 or 0.10.8 of System?

It may be specific to the upgrade scenario.  That remains to be seen.  If you can also compare the versions and if you see the error on a non-upgrade stack scenario it would be helpful for us to triage faster.

I acknowledge in this particular case the bug relates to packages, which makes it harder.  We are aware of the system - metricset load failure bug.  I can confirm, this may still be seen on some older systems, but we have not been able to find a cause, and had all decided thus far not to fix it or attempt further, so we can put a Docs bug in.  But if you are using Windows 10, it should work, probably and we need to follow up.

That may be causing the unhealthy agent, but maybe not.  So, are the 2 issues related?  we aren’t sure.  

About the ‘unhealthy’ agent, we are still seeing that in 7.11 and 8.0 - I think it is still a real bug, but is it critical?   The hard part is that the state of the Agent as 'unhealthy' may be ‘correct’ initially - but after 5+ minutes we think it is likely a bug if things are working but it stays as such.

**Comment by amolnater-qasource** [comment]
Hi, @EricDavisX 
We have revalidated this issue on 7.11.0 Kibana cloud environment(upgraded from 7.10.2).

> which (specific) windows OS this is on?

Tested on Windows 10 x64

> And which version of the package is on the system you are testing - is it 0.10.7 or 0.10.8 of System?

After upgrading the Kibana cloud environment, System integration version: 0.10.7

>  if you see the error on a non-upgrade stack scenario

**Scenario 1:**
There is no "Metricbeat error" on non-upgrade stack scenario. This issue is only specific to Kibana upgrade.

**Scenario 2:**
When a 7.11 BC-5 agent is installed with **pre-existing default policy( 7.10.2)** after kibana upgrade.

**What's not working:**
- "error: Metricbeat FAILED" errors under Logs tab.
- No data for system integration under Data streams tab.
- Permanent "Unhealthy" status.

**Screenshots:**
![Upgrade](https://user-images.githubusercontent.com/77374876/106748854-d52cde80-664b-11eb-8272-b5ee100c2085.png)

**What's working:**
When 7.11 BC-5 agent is installed **with newly created policy** after kibana upgrade.

**Observations:**
- Data for system integration under datastreams tab.
- No "Metricbeat" errors observed under Logs tab.
- Agent status is "Healthy" throughout.

Please let us know if anything else is required.
Thanks 
QAS

**Comment by EricDavisX** [comment]
Thank you @amolnater-qasource - that is very helpful.
@ph this is therefore going to be found by anyone using 7.10.x who upgrades to 7.11 and keeps their deployments  / data and Agents. The work-around is to create a new policy and re-deploy Agent.  

Every release we do I expect more users to try it and to keep it around and upgrade it - while we are still Beta it may be worth fixing for 7.11.1 if not the .0 release right now.   And it would be good to know the technical problem we've run into, before we plan where the fix would go.  Can we assign someone?

@ruflin FYI, too.  And it is another Fleet impacting stack upgrade failure, justifying spending time on the upgrade test we started tracking recently.

**Comment by EricDavisX** [comment]
i confirmed this was seen as passing in 7.11 BC3 testing a few weeks ago - link: https://elastic.testrail.io/index.php?/runs/view/1005&group_by=cases:section_id&group_order=asc&group_id=9194
 - not sure what changed in the last approximately 4 weeks that relates?

**Comment by ph** [comment]
Let me try to understand I think @ruflin could confirm this.

The scenario described here is the following.

- Start with 7.10
- Have the Agent enrolled into a Policy with the System integration.
- Agent are healthy
- Upgrade to 7.11
- Upgrade agent to 7.11
- Agent are becoming unhealthy.


**What we need to identify here is:**

1. It is a problem with Elastic Agent after the upgrade? Most likely?
2. it is a problem with the Integration itself? Less likely because the content of the agent should be same as before.
3. It is a problem in Fleet on upgrade of the package? Less likely.

From @EricDavisX  or @amolnater-qasource can you add Metricbeat log in this issue. In the Elastic Agent we see that metricbeat is restarted this mean that the error is fatal and the agent try to recover.


@michalpristas I've put this in investigate from triage and we can do it in the 7.13 iteration.
 Anything you could think of here?

Impact, agree with your evaluation for 7.11.1, I think it's too late for 7.11.0 and there is a workaround.

**Comment by ph** [comment]
@EricDavisX as bad as it look I would not consider this a blocker. Do we know if the reassigning of an agent to another Policy fixes the issues? I am trying to understand where the problem could be located.

**Comment by EricDavisX** [comment]
it looks to me like assigning to a new policy fixes it, indeed - I don't know about re-assigning to an existing policy.  No data on that yet, but if the problem is related to the Integration version then an existing policy wouldn't fix it.

**Comment by ph** [comment]
We can try to reproduce but to really speed up the investigation we need this.

1. Metricbeat logs, if it restarts every minute the log should be noisy.
2. Would be great to see the Agent Policy YAML from the Elastic Agent side (action_store.yml) and from Kibana export yaml.

**Comment by EricDavisX** [comment]
I am ok with it not being in 7.11.0 - can we site it as a 'known issue' wherever appropriate?  not sure the logistics for that.

**Comment by fearful-symmetry** [comment]
Chiming in here, since the `system/load` issues on Linux are just a thorn in my side. Nothing in the change from 0.10.7 to 0.10.8 should impact `system/load` so that shouldn't matter. Can we verify  _before the upgrade to 7.11_, what version of the policy is in place? If for whatever reason 7.10 is spinning up with an older integration release, this would explain it, although I'm not sure how that would happen.

**Comment by fearful-symmetry** [comment]
Also seconding PH, we/I need the metricbeat logs, as well as the `action_store.yml`.

**Comment by amolnater-qasource** [comment]
Hi @ph 
As per feedback on https://github.com/elastic/beats/issues/23812#issuecomment-772638580 
> Upgrade agent to 7.11

We are installing new 7.11 agent with old existing default policy(7.10.1). **No agent upgrading is involved.**

**Steps to reproduce:**
- Start with 7.10.2 which will have **its own default policy** with system integration (0.9.1).
- Install an agent on 7.10.2 with default policy.
- Upgrade Kibana to 7.11.0
- Install 7.11 agent to  **pre existing default policy** with system integration(upgraded to 0.10.7)  and **not the new policy** created manually after Kibana upgrade.
- Observe Agent is unhealthy and "Metricbeat" errors under Logs tab.

Build details before Kibana Upgrade | Build details after Kibana Upgrade
-- | --
BUILD: 36136 COMMIT: a0b793698735eb1d0ab1038f8e5d7a951524e929 System Version: 0.9.1 | BUILD: 37827 COMMIT: d801a7bb08e17368584d00fbd97a5d0006285b51 System Version: 0.10.7



**Logs:**
[elastic-agent-json.log](https://github.com/elastic/beats/files/5924379/elastic-agent-json.log)
[metricbeat_monitor-json.log](https://github.com/elastic/beats/files/5924381/metricbeat_monitor-json.log)
[action_store_yml.txt](https://github.com/elastic/beats/files/5924509/action_store_yml.txt)
[elastic_agent_yml.txt](https://github.com/elastic/beats/files/5924516/elastic_agent_yml.txt)
[metricbeat_log.txt](https://github.com/elastic/beats/files/5924531/metricbeat_log.txt)

**UI logs:**
[UI_agent_logs.txt](https://github.com/elastic/beats/files/5924607/UI_agent_logs.txt)

Please let us know if anything else is required.
cc: @EricDavisX @fearful-symmetry 
Thanks 
QAS

**Comment by amolnater-qasource** [comment]
Hi @ph 
As per feedback on https://github.com/elastic/beats/issues/23812#issuecomment-772638580 
> Upgrade agent to 7.11

We are installing new 7.11 agent with old existing default policy(7.10.1). **No agent upgrading is involved.**

**Steps to reproduce:**
- Start with 7.10.2 which will have **its own default policy** with system integration (0.9.1).
- Install an agent on 7.10.2 with default policy.
- Upgrade Kibana to 7.11.0
- Install 7.11 agent to  **pre existing default policy** with system integration(upgraded to 0.10.7)  and **not the new policy** created manually after Kibana upgrade.
- Observe Agent is unhealthy and "Metricbeat" errors under Logs tab.


Build details before Kibana Upgrade | Build details after Kibana Upgrade
-- | --
BUILD: 36136 | BUILD: 37827
COMMIT: a0b793698735eb1d0ab1038f8e5d7a951524e929 | COMMIT: d801a7bb08e17368584d00fbd97a5d0006285b51
System Version: 0.9.1 | System Version: 0.10.7


**Logs:**
[elastic-agent-json.log](https://github.com/elastic/beats/files/5924379/elastic-agent-json.log)
[metricbeat_monitor-json.log](https://github.com/elastic/beats/files/5924381/metricbeat_monitor-json.log)
[action_store_yml.txt](https://github.com/elastic/beats/files/5924509/action_store_yml.txt)
[elastic_agent_yml.txt](https://github.com/elastic/beats/files/5924516/elastic_agent_yml.txt)
[metricbeat_log.txt](https://github.com/elastic/beats/files/5924531/metricbeat_log.txt)

**Kibana UI logs:**
[UI_agent_logs.txt](https://github.com/elastic/beats/files/5924607/UI_agent_logs.txt)

Please let us know if anything else is required.
cc: @EricDavisX @fearful-symmetry 
Thanks 
QAS

**Comment by ph** [comment]
OK, I've looked at the log, We saw in the kibana ui screen the issues this is indeed the following:

```
{"log.level":"info","@timestamp":"2021-02-04T03:51:17.564-0500","log.logger":"centralmgmt.fleet","log.origin":{"file.name":"fleet/manager.go","file.line":138},"message":"Status change to %!s(management.Status=5): 1 error: 1 error: Error creating runner from config: 1 error: metricset 'system/load' not found","ecs.version":"1.5.0"}
```
When Elastic agent tries to create the input it will fail because of the missing metric set, this will do make a status `Status == 5`, 5 == Failed, on failed elastic agent restart the processes.

I believe in that case the Agent does the right thinig, we delegate the input creation to metricbeat and metricbeat return an error. For me it's a problem with either the integration, the metricset or metricbeat,

 @fearful-symmetry I am going to reassign that to you.

I am making a PR to make the error human readable.

**Comment by EricDavisX** [comment]
@ph Thank you. @fearful-symmetry and I were already reviewing other open issue about the metricset 'system/load' conditional.  The package seems to have the correct condition, unless it isn't using the Agent's spec'ed functionality correctly.  I looked in slack and saw a post from @blakerouse mentioning we probably want / need the condition 'on the input' in this case, but I didn't know how to interpret that against what we already had.  It is great to have a lead tho, it is super that it can be fixed in the package and released out of band if we wanted!

**Comment by ph** [comment]
@fearful-symmetry reading your comment again make me push a bit more.

> Chiming in here, since the system/load issues on Linux are just a thorn in my side. Nothing in the change from 0.10.7 to 0.10.8 should impact system/load so that shouldn't matter. Can we verify before the upgrade to 7.11, what version of the policy is in place? If for whatever reason 7.10 is spinning up with an older integration release, this would explain it, although I'm not sure how that would happen.

I think you have pointed to the issue, this is what is happening..

7.10

- Fleet starts.
- Fleet download the latest version of the system integration (version 1).
- Fleet create a Default Agent Policy.
- Fleet create a system integration policy in the agent (version 1), this will be a snapshot of the current configuration.

When upgrading to 7.11.
- Fleet upgrade the System integration to the most recent version.
- Agent Policy are not updated.
- System integration policy in the agent will still be the version 1.

If you create a new Agent policy and add the system integration it will be set to version 2.

What is not clear to me is, why it work with 7.10, it should not have worked. Why it has an older version of the system package.

If you make any edit to  the System integration policy it should update it to the latest version.

**Comment by ph** [comment]
Looking at the `action_store` the system package is really old and this version has the `system/load` issue.

````
    meta:
      package:
        name: system
        version: 0.9.1
```

Why this version is installed, is there kibana conditional that would prevent an update? 
@elastic/fleet would be nice to have your input here.

**Comment by fearful-symmetry** [comment]
> What is not clear to me is, why it work with 7.10, it should not have worked. Why it has an older version of the system package.

Yah, that's what I'm not sure about either. I've seen this in the past, before we had the dynamic config templates, but the issue was (I think) limited to an `unhealthy` status with no data issues.


> Why this version is installed, is there kibana conditional that would prevent an update?

No idea there either. https://epr.elastic.co/search?package=system says `0.10.7`.

> WORKAROUND: If you make any edit to the System integration policy it should update it to the latest version and fix the problem.

This is the biggest issue, IMO, and it's once that's shot in me foot a few times just casually developing--Kibana doesn't make it clear what version of an Integration is actually _in_ a policy, and there isn't a way to upgrade it. At minimum, all the integrations in a policy should have a little tag that displays the version number.

**Comment by ph** [comment]
>  and there isn't a way to upgrade it.
There is a way, but not obvious to the user, any edit will update the integration policy.

The problem is the upgrade of the integration and the update of the integration policy aren't linked, we did by design, but the behavior is not the same endpoint will update the endpoint policy.

> all the integrations in a policy should have a little tag that displays the version number.

@mostlyjason there is a lot to consume in this thread, but something to consider, maybe in the medium term.

**Comment by fearful-symmetry** [comment]
> There is a way, but not obvious to the user, any edit will update the integration policy.

Ooof, yah, that's a tad non-obvious. 

I'm still worried about how they got `0.9.1`. If. user is running that integration against 7.11, there isn't much I can do. At this point we should consider it a buggy release, since that was pre-dynamic config. However, none of the EPRs seem to report that package.

**Comment by mostlyjason** [comment]
> all the integrations in a policy should have a little tag that displays the version number.

++ that is a great idea. I opened an issue to track this here https://github.com/elastic/kibana/issues/90308

**Comment by EricDavisX** [comment]
To @fearful-symmetry 's note, when `0.9.1` was released to prod storage EPR, a user could have created a policy with it at that time, and not have updated it since, making that policy frozen at that point in time.  I can cite that I have a 7.10.1 cloud stack that is showing the 0.9.1 System Integration as the latest available, currently - so that is helpful as a reference. 
 
What does a user need to do to get the latest packages pulled from the registry into Kibana - I'm thinking maybe  it requires a restart of Kibana, and I question if this is a design we can stick with.

**Comment by EricDavisX** [comment]
Update:  I restarted by 7.10.1 Kibana cloud instance, and it still shows .0.9.1 Integration, which surprised me a little from the above discourse.  I poked in to Integrations repo and find that 7.10.x is pinned to 0.9.2 System version, at which point we jump to 0.10.x with a reference to ^7.11.0 for the Kibana version.  

I see 0.9.1 still exists in prod registry as [here](https://epr.elastic.co/search?package=system&all=true). 
And the increment to 7.11 Kibana requirement for System is[here](https://github.com/elastic/integrations/commit/2b4f96bd0760e9639a35ec7c27dc2c90f6b466c7#diff-d4cd9d386b49496970c932d312ae09b5a2acc2c3f85f75a7819064d67634248b)

So, this nearly guarantees the noted problem will be seen unless users are well informed on upgrade nuances, which I think we may not have covered well yet?  The answer for now is in Docs or in reconsidering pushing a System package for 7.10.x that has support for conditional logic for the offending metricset.  

For docs... we would want to document that a user needs to update the package policy and Agent policy before manually installing new Agent or clicking 'upgrade' Agent button after upgrading the stack.  @ph and @fearful-symmetry thoughts?

**Comment by fearful-symmetry** [comment]
>  I poked in to Integrations repo and find that 7.10.x is pinned to 0.9.2 System version, 

Alright, that makes sense. Until we have a better mechanism for users to upgrade policies, we'll need some kind of documentation. Presumably adding a snippet to the upgrade docs will do fine?

**Comment by ph** [comment]
Thanks, @EricDavisX and @fearful-symmetry for confirming the theory.

I agree that we should add a doc snippet detailing the problem, @EricDavisX Can you do that?

@fearful-symmetry does the suggestion of creating a `0.9.2` package possible its that something we should consider?

**Comment by fearful-symmetry** [comment]
> does the suggestion of creating a 0.9.2 package possible its that something we should consider?

@ph So, the issue has more to do with the support for dynamic config. Someone will need to confirm this, but I'm about 70% sure that the dynamic input stuff isn't in 7.10, which I assume is what we'd want to target with `0.9.2`.

**Comment by ph** [comment]
@fearful-symmetry we have targetted that in 7.10 https://github.com/elastic/beats/pull/20994

**Comment by fearful-symmetry** [comment]
Oh, awesome! So is the goal just to push this for people staying on 7.10?

**Comment by ph** [comment]
Yes, we have to tests to make sure it work, but that would reduce the problem.

**Comment by EricDavisX** [comment]
About the reason:  It would reduce the pain on 7.11 users who upgrade from 7.10.  Which helps us in the long term avoid SDH / Discuss support issues, etc.

**Comment by amolnater-qasource** [comment]
Hi @ph 
As per your feedback on https://github.com/elastic/beats/issues/23812#issuecomment-773287986 , we have revalidated this issue.
> System integration policy in the agent will still be the version 1

Yes, this is correct, as per our observations:

- Default policy System integration remains v0.9.1 after Kibana upgrade(7.10.2->7.11.0)
- Newly created policy is created with Latest system integration and gives no error with agent enrollment.

> Kibana doesn't make it clear what version of an Integration is actually in a policy, and there isn't a way to upgrade it.

For this, @fearful-symmetry to check the version of  installed integrations in a policy on Kibana UI :
We can navigate to <Policy's> Actions button and select "View Policy".

Policies>default policy>Click "View Policy".

**Screenshots:**
![Upgrade Issue](https://user-images.githubusercontent.com/77374876/107025359-5496eb00-67cf-11eb-8948-3655e0ebc066.png)

Further as per:
> WORKAROUND: If you make any edit to the System integration policy it should update it to the latest version and fix the problem.

@ph we have attempted to update the name of System integration by renaming it under default policy. However, we didn't observe any version change for System integration. Also, no change agent behaviour.

Please let us know if we have missed anything. 

Thanks 
QAS

**Comment by fearful-symmetry** [comment]
@ph just realized that we don't seem to have any system for pushing to the equivalent of `9.x`. Can we just push to `package-storage` directly?

**Comment by ph** [comment]
Not sure better to check with  @mtojek or @ycombinator ?

**Comment by ycombinator** [comment]
> just realized that we don't seem to have any system for pushing to the equivalent of `9.x`. 

I caught up on the thread but I'm a bit lost by the `9.x`? @fearful-symmetry do you mind elaborating a bit on what you'd like to do?

**Comment by fearful-symmetry** [comment]
@ycombinator as in, if I wanted to push an update to `0.9.3` (as `0.9.4`) as opposed to `0.10.9`, how would I do that?

**Comment by ycombinator** [comment]
Chatted with @fearful-symmetry off-issue to understand what we want to achieve here. Long story short, we tried putting up a PR to create a `0.9.3` version of the `system` package with the `condition` change in the `integrations` repo but ran into some CI issues (which are to be expected, TBH): https://github.com/elastic/integrations/pull/674.

An alternative that would work is to make a PR for the `0.9.3` version of the `system` package directly to the `package-storage` repo's `snapshot` branch but this has a couple of downsides:
- we'd bypass all the checks that would normally happen on package changes in the `integrations` repo, and
- we wouldn't have any record of the `0.9.3` changes in the `integrations` repo.

We were trying to avoid these downsides with https://github.com/elastic/integrations/pull/674 but, as mentioned above, we're running into some issues with CI there.

@EricDavisX @ph how urgently is this change needed? If it could wait a day or so, I'd like to get @mtojek's thoughts on it here: https://github.com/elastic/integrations/pull/674#issuecomment-775398748.

**Comment by ycombinator** [comment]
Thanks @EricDavisX @ph. We might have a way forward (see https://github.com/elastic/integrations/pull/674#issuecomment-775429821) but I'd still like @mtojek to weigh in on that as well.

**Comment by ycombinator** [comment]
`system-0.9.3` has been released. It is currently available in the `snapshot` package registry. I will leave it up to @fearful-symmetry or someone else on this thread to promote it to `staging` and `production` package registries as and when it's appropriate to do so.

Like its predecessor, `system-0.9.2`, the `system-0.9.3` package it is intended to work with Kibana `^7.10.0`. But unlike `system-0.9.2`, the `load` dataset in `system-0.9.3` will only work when `${host.platform} != 'windows'`.

**Comment by fearful-symmetry** [comment]
`0.9.3` is currently in the staging registry.

**Comment by ph** [comment]
thanks @fearful-symmetry and @ycombinator,

@EricDavisX we should be able to test the change in the upgrade scenario.

**Comment by EricDavisX** [comment]
@ph acknowledged.  I'm not sure how to run a self-managed stack upgrade but we can look for docs, and we can test the package in snapshot repo.  And if it is good then upgrade it to prod and test via cloud (the precise exact same scenario). we'll get it done.

**Comment by fearful-symmetry** [comment]
Yah, not sure how exactly we should test this either...

**Comment by EricDavisX** [comment]
the test team knows how to (theoretically) run a self managed upgrade, but none of us are practiced on it.  they attempted, and ran into errors - so while we triage and track that, we can just test the package and if passing muster we can run the upgrade test via cloud as we had prior, after merging the integration up.  

I'll broadcast test progress as we get it and we can coordinate here.

**Comment by EricDavisX** [comment]
I believe the team has tested, allow me to validate and we can promote the package (and test again 7.10)

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on self managed stack upgraded to 7.11.1 BC-2  Kibana environment(from 7.10.2 Snapshot) and found it **still reproducible**.

**Steps followed:**
1. Deploy self managed 7.10.2 Snapshot Kibana environment with below xpack entry in kibana.yml
`
xpack.ingestManager.registryUrl: "http://epr-snapshot.elastic.co"
`
2. Observe System integration version 0.9.2 is shown up on accessing Fleet UI.
3. Upgrade 7.10.2 Snapshot Kibana to 7.11.1 by replacing config & data folders with older folders present under extracted elasticsearch & kibana folders.
4. Re-run elasticsearch.bat and kibana.bat files.

**Observations:**
- Error message "error: Metricbeat Failed" is showm in Agent logs in Fleet UI when 7.11.1 agent is installed with pre-existing default policy.
- Pre-existing Default policy is still having system integration version as 0.9.1 (No impact of xpack entry).
Further, on updating xpack entry to `xpack.fleet.registryUrl: "http://epr-snapshot.elastic.co"` for upgraded 7.11.1 Kibana
and we observed security errors under Fleet>Agents tab.

**Screenshot:**
![4](https://user-images.githubusercontent.com/77374876/108028680-37c1a980-7052-11eb-9d81-2ffee5622d9c.PNG)

**Build details:**
```
Artifact link used: https://staging.elastic.co/7.11.1-bc1b1b1c/summary-7.11.1.html
Build: 37941
Commit: 348233f89825946d65101f6d9082567353459b8e
```
**Note:**
This issue is not occurring with Linux and mac agents on upgraded 7.11.1 Kibana cloud environment using previously existing policy (from 7.10.2).

Thanks
QAS

**Comment by EricDavisX** [comment]
the above description is helpful - it highlights a things we can follow up and a significant failure to provide you with all of the steps required and working code to prove out the changes.  It is a lot going on actually, but I'll try to itemize clearly what I see and who can help follow up.

1) It is reported that the 0.9.1 System package was baked in to the Default policy, this is most likely explained by the Fleet 'setup' call being done before the 'snapshot' xpack kibana.yml value was changed, and so the production EPR storage was still in use and it points to 0.9.1 for now.  We need to run a test and ensure that we do NOT start Kibana before changing the value (on an entirely fresh system).  

... And so in this case, if it still fails to bake in 0.9.2 (or 0.9.3) to the Default policy, this is a bug in Kibana Fleet side... which would seem to indicate Kibana is querying the prod storage to set up policy even though there is a custom registry url set.  Custom Registry is not a feature we support now but it is in the future plans so would be good to document (and indeed we are using it during dev to test it out).  Now that we know about this we can modify our testing by creating a new policy to use as our 'base' test before we upgrade, we should ensure the new policy has the new package we want to test.  Let me log this separately as a follow up, here: 
https://github.com/elastic/kibana/issues/91502

2) Package storage 'snapshot' branch of the repo is in use and yet still the .0.9.2 System package is showing in the Kibana UI instead of 0.9.3.  The 0.9.3 version is available in the actual registry API call:
https://epr-snapshot.elastic.co/search?package=system&all=true

when I look in the storage 'snapshot' branch...
https://github.com/elastic/package-storage/tree/snapshot/packages/system
 this shows for me only the 0.9.2, but it was *just merged around 8:30 AM today that the 0.9.3 would be removed to promote it to 'staging' storage.  so I'm not sure where our process breaking down here... the snapshot repo should have

the snapshot manifest shows the 0.9.3 package is tied to 7.10, as it should be:
https://github.com/elastic/package-storage/blob/snapshot/packages/system/0.9.3/manifest.yml#L13 

and I confirmed we rolled out of the package storage cluster with 'snapshot' before testing on Feb 15th as:
https://beats-ci.elastic.co/job/Ingest-manager/job/release-distribution/47/

perhaps this is a caching issue?  Or the environment was not
I have split this off as a separate concern here: 
https://github.com/elastic/kibana/issues/91505

Since the .9.3 System package didn't show up in Kibana per your test, even tho we thought it should have, this  means we'll need to re-run from the very beginning when we get the 0.9.3 'showing up in kibana' problem solved.  @amolnater-qasource @dikshachauhan-qasource fyi - We desire to get the new 0.9.3 System package set up in a policy, we can run the detailed fields test with it in 7.10.x against Linux, Windows, macOS to validate the package, we can report back separately, outside of this ticket please (to keep scope contained). This ticket can remain open for the usage of he conditionals in the System package which we hope will avoid the 'Metricbeat FAILED' errors in the logs and any problems with the running processes.  The progress is blocked until we can get # 2 above resolved. 

@ph do we have anyone available to help with # 2 first, and then # 1 above, perhaps @skh knows this area fairly well?

**Comment by EricDavisX** [comment]
@amolnater-qasource if you have the disable.TLS setting in place in your kibana.yml and you are seeing the error
![image](https://user-images.githubusercontent.com/12970373/108086952-ebbc3800-7044-11eb-858c-34515d3e2179.png)
 - please log it as a separate bug, blocking our progress here, thank you for noting it - but it is helpful not to make the comments logs any longer than they have to be.

**Comment by ph** [comment]
@EricDavisX thanks for the update, I also checked on my side that the system on the snapshot is indeed 0.9.3.. So maybe its a fleet EPM issues, @skh or @jfsiii should be able to help with area of code.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on self-managed stack upgraded to 7.11.1 BC-3 Kibana environment(from 7.10.3 Snapshot) and found it **fixed** for self-managed stack upgrade.

**Steps followed:**
1. Deploy self managed 7.10.3 Snapshot Kibana environment with below xpack entry in kibana.yml
`
xpack.ingestManager.registryUrl: "http://epr-snapshot.elastic.co"
`
2. Observe System integration version 0.9.3 is shown up on accessing Fleet UI.
3. Create a new policy having system integration version 0.9.3 
3. Upgrade 7.10.3 Snapshot Kibana to 7.11.1 by replacing config & data folders with older folders present under extracted elasticsearch & kibana folders.
4. Re-run elasticsearch.bat and kibana.bat files.
5. Install 7.11.1 BC-3 agent with new policy(system version 0.9.3)

**Observations:**
- No "error: Metricbeat Failed" observed under agent Logs tab.
- Agent status is "Healthy".

**Screenshots:**
![Upgrade](https://user-images.githubusercontent.com/77374876/108182550-226e7d00-712f-11eb-8b00-41a0dc578996.png)

We will close this issue as soon as it is available for cloud build.
Please let us know if anything else is required. 

Thanks
QAS

**Comment by EricDavisX** [comment]
ok great, so we will pr and push system 0.9.3 to prod

**Comment by EricDavisX** [comment]
it is ready for test on 7.10.x latest shipped version!

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on upgraded 7.11.1 Kibana cloud environment(from 7.10.2 release build) and found this issue **fixed**. Hence closing this out.

**Screenshot:**
![6](https://user-images.githubusercontent.com/77374876/108365726-8c655000-721d-11eb-9aa2-5e758165e9ea.JPG)

Please let us know if anything else is required.
Thanks 
QAS
